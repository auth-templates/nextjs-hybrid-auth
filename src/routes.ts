export const PrivateRoutes = {
    dashboard: '/dashboard',
    account: '/account',
    leaves: '/leaves'
};

export const PublicRoutes = {
    login: '/login',
    register: '/register',
    recoverPassword: '/recover-password',
    resendConfirmationEmail: '/resend-confirmation-email',
    resetPassword: '/reset-password/:token',
    verifyAccount: '/verify-account/:token',
    twoStep: '/two-step',
}