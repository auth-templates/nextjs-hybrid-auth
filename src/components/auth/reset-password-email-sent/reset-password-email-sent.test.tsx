import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ResetPasswordEmailSent from './reset-password-email-sent';
import { PublicRoutes } from '../../../routes';

describe("ResetPasswordEmailSent", () => {
    it('should contain all elements', async () => {
        render(
            <MemoryRouter>
                <ResetPasswordEmailSent email={'account@mail.com'} />
            </MemoryRouter>
        );

        expect(screen.getByText(/An email with reset password intructions has been sent to/)).toBeInTheDocument(); 
        expect(screen.getByText(/account@mail.com/)).toBeInTheDocument();
        expect(screen.getByText(/and it should arrive soon. Please check your inbox./)).toBeInTheDocument();
    });

    it("renders correctly", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <ResetPasswordEmailSent email={'account@mail.com'} /> 
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});