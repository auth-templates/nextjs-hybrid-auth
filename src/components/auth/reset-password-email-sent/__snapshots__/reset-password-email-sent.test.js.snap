// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordEmailSent renders correctly 1`] = `
<div
  className="resetPasswordEmailSent"
>
  An email with reset password intructions has been sent to
  <div
    className="email"
  >
    account@mail.com
  </div>
  and it should arrive soon. Please check your inbox.
</div>
`;
